pyomo/repn/plugins/standard_form.py

```diff
@@ -0,0 +1,537 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import collections
+import logging
+from operator import attrgetter
+
+from pyomo.common.config import (
+    ConfigBlock,
+    ConfigValue,
+    InEnum,
+    document_kwargs_from_configdict,
+)
+from pyomo.common.dependencies import scipy, numpy as np
+from pyomo.common.gc_manager import PauseGC
+from pyomo.common.timing import TicTocTimer
+
+from pyomo.core.base import (
+    Block,
+    Objective,
+    Constraint,
+    Var,
+    Param,
+    Expression,
+    SortComponents,
+    Suffix,
+    SymbolMap,
+    maximize,
+)
+from pyomo.opt import WriterFactory
+from pyomo.repn.linear import LinearRepnVisitor
+from pyomo.repn.util import (
+    FileDeterminism,
+    FileDeterminism_to_SortComponents,
+    categorize_valid_components,
+    initialize_var_map_from_column_order,
+    ordered_active_constraints,
+)
+
+### FIXME: Remove the following as soon as non-active components no
+### longer report active==True
+from pyomo.core.base import Set, RangeSet, ExternalFunction
+from pyomo.network import Port
+
+logger = logging.getLogger(__name__)
+
+RowEntry = collections.namedtuple('RowEntry', ['constraint', 'bound_type'])
+
+
+# TODO: make a proper base class
+class LinearStandardFormInfo(object):
```

Comment:
Just an FYI: You no longer need to do `class blahBlah(object):` in Python 3. You can just do `class blahBlah:`.

Reply:
I always like explicitly declaring base classes (plus that makes it consistent with the rest of Pyomo, which all still explicitly lists `object`)

Reply:
Yeah but John, it'll be the first thing that `pylint` yells at you to refactor, should we ever decide to think about incorporating static code analysis nonsense.